aws/terraform/modules/metaflow/modules/metadata-service/ec2.tf

```diff
@@ -0,0 +1,67 @@
+resource "aws_security_group" "metadata_service_security_group" {
+  name        = local.metadata_service_security_group_name
+  description = "Security Group for Fargate which runs the Metadata Service."
+  vpc_id      = var.metaflow_vpc_id
+
+  ingress {
+    from_port   = 8080
+    to_port     = 8080
+    protocol    = "tcp"
+    cidr_blocks = [var.vpc_cidr_block]
+    description = "Allow API calls internally"
+  }
+
+  ingress {
+    from_port   = 0
+    to_port     = 0
+    protocol    = -1
+    self        = true
+    description = "Internal communication"
+  }
+
+  # egress to anywhere
+  egress {
+    from_port   = 0
+    to_port     = 0
+    protocol    = "-1" # all
+    cidr_blocks = ["0.0.0.0/0"]
+    description = "Allow all external communication"
+  }
+
+  tags = merge(
+    var.standard_tags,
+    {
+      Metaflow = "true"
+    }
+  )
+}
+
+resource "aws_lb" "this" {
+  name               = "${var.resource_prefix}nlb${var.resource_suffix}"
```

Comment:
I had to double check, but the decision for the ALB vs NLB was made by simply using what the original `metaflow-tools` Cloud Formation stack leveraged:

https://github.com/Netflix/metaflow-tools/blob/master/aws/cloudformation/metaflow-cfn-template.yml#L192

Now that we have the entire stack in Terraform, I think it would make sense to look through the stack and see what changes we could make.

As for specifically ALB vs NLB in this context, I don't believe I'm in a position to comment one way or the other.

Reply:
Happened to glance at this convo, I can contextualize it.  ALB is probably a better choice conceptually, but this LB exists almost exclusively to provide a pathway for API Gateway to have a secure route to the MDS living in Fargate so it can terminate TLS and overlay basic auth keying.  At the time, only REST API's were available in APIGW, and the VPC Link functionality for the had a requirement to forward the connection to an NLB.  APIGW's newer HTTP framework does things slightly differently, which seems to support ALB with a bit of fiddling, but we'd need to refactor the frontend API.  Probably not a bad idea, but likely a bigger effort than we should tackle here.